src/mdesktopentry.cpp

```diff
@@ -243,32 +288,51 @@ QString MDesktopEntry::fileName() const
 
 bool MDesktopEntry::contains(const QString &key) const
 {
+    if (key.startsWith(DesktopEntrySection + "/")) {
+        return d_ptr->desktopEntriesMap.contains(key.mid(DesktopEntrySection.length() + 1));
+    }
     return d_ptr->desktopEntriesMap.contains(key);
```

Comment:
True. Let's see below if we want to have other content or not, I'm open for both options.